app/models/concerns/state_machineable.rb

```diff
@@ -0,0 +1,37 @@
+require 'active_support/concern'
+
+module StateMachineable
+  extend ActiveSupport::Concern
+
+  included do
+    after_initialize :initialize_state
+
+    class_attribute :state_attribute
+    class_attribute :state_machine_class
+  end
+
+  def reload(options = nil)
```

Comment:
My main concern with this is that we won't return what's persisted in the database but rather our state machine current state (if for instance the state is `nil`) This might lead to some unexpected behaviour. Just wondering why we needed to add this for journeys, was it for testing purposes?

Reply:
Yes, this is just to support the specs - we don't have any code that calls `reload` outside of the test suite. This was particularly affecting the event runner for journeys - the runner actions state machine changes correctly but when the spec calls e.g. `journey.reload` the object under test was fetching the updated state attribute but not updating the state machine accordingly.

ActiveRecord has an `after_find` callback that would seem suited for this, but it's not called as part of `reload`. The alternative to this would be explicitly finding objects again within the test suite which is a bit of a pain, so this seems an acceptable approach to mitigate that.